proxy.go

```diff
@@ -47,15 +47,39 @@ var sandboxID string
 
 var proxyLog = logrus.New()
 
+// This function is meant to run in a go routine since it will send ping
+// commands every second. It behaves as a heartbeat to maintain a proper
+// communication state with the Yamux server in the agent.
+func heartBeat(session *yamux.Session) {
+	if session == nil {
+		return
+	}
+
+	for {
+		if session.IsClosed() {
+			break
+		}
+
+		session.Ping()
+
+		// 1 Hz heartbeat
+		time.Sleep(time.Second)
```

Comment:
Just noting, I suspect 1s might be a good balance between too much traffic and keeping the link alive (waking out of freeze). For reference, I believe the default keepAlive timing we are effectively replacing was 30s.

Reply:
my biggest concern here is how much overhead add this?

Reply:
What do you mean by overhead ? Amount of data we're sending to the agent ?

Reply:
I get the cpu usage, but could you elaborate a bit more on memory consumption and bottle necks ?

Reply:
I ran some tests and lgtm